Q42/accessibility-data-pipeline
# Update schema and upgrade query builders

# Description

The requested started with the addition of new properties to the schemas. This involved update the struct fields in bigquery which complicated querying. The new fields didn't adhere to the order of fields dictated by the utilised json schema. To fix this issue and subsequently improve all querying fields are queried using the order present in the schema at the moment of running the pipeline. This way the correct order of fields is guaranteed.

This has currently only been implemented for android.

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

To test this I created a copies of the event and aggregated tables. These copies were used to test against.
The steps are as follows:
- Copy tables as explained in the schemas readme
- Update these table with the new schema
- Run the pipeline job using the original firestore collection but with the newly created event and aggregated tables. Make sure to also disable the firestore cleanup step to prevent loss of data that is to be processed in the nightly running job.

Working test run in Vertex AI which was executed locally: 
`local-run-android-test-v3-2023-03-10-1678464437`
